noxfile.py

```diff
@@ -183,6 +183,75 @@ def system(session):
         )
 
 
+@nox.session(python=DEFAULT_PYTHON_VERSION)
+def prerelease(session):
+    session.install(
+        "--prefer-binary",
+        "--pre",
+        "--upgrade",
+        "alembic",
+        "geoalchemy2",
+        "google-api-core",
+        "google-cloud-bigquery",
+        "google-cloud-bigquery-storage",
+        "google-cloud-core",
+        "google-resumable-media",
+        "grpcio",
+        "sqlalchemy",
+        "shapely",
+    )
+    session.install(
+        "freezegun",
+        "google-cloud-testutils",
+        "mock",
+        "psutil",
+        "pytest",
+        "pytest-cov",
+        "pytz",
+    )
+
+    # Because we test minimum dependency versions on the minimum Python
+    # version, the first version we test with in the unit tests sessions has a
+    # constraints file containing all dependencies and extras.
+    with open(
+        CURRENT_DIRECTORY
+        / "testing"
+        / f"constraints-{UNIT_TEST_PYTHON_VERSIONS[0]}.txt",
+        encoding="utf-8",
+    ) as constraints_file:
+        constraints_text = constraints_file.read()
+
+    # Ignore leading whitespace and comment lines.
+    deps = [
+        match.group(1)
+        for match in re.finditer(
+            r"^\s*(\S+)(?===\S+)", constraints_text, flags=re.MULTILINE
+        )
+    ]
+
+    # We use --no-deps to ensure that pre-release versions aren't overwritten
+    # by the version ranges in setup.py.
+    session.install(*deps)
+    session.install("--no-deps", "-e", ".[all]")
```

Comment:
Is `[all]` an extra we define here? I thought that was just a `google-cloud-bigquery` thing?